pennylane/templates/subroutines/qdrift.py

```diff
@@ -0,0 +1,227 @@
+# Copyright 2018-2021 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Contains template for QDrift subroutine.
+"""
+import pennylane as qml
+from pennylane.operation import Operation
+from pennylane.ops import Sum
+import numpy as np
+
+class QDrift(Operation):
+    r"""An operation representing the QDrift subroutine for the complex matrix exponential
+    of a given Hamiltonian.
+
+    The QDrift subroutine provides a method to approximate the matrix exponential of hamiltonian
+    expressed as a linear combination of terms which in general do not commute. Consider the hamiltonian
+    :math:`H = \Sigma^{N}_{j=1} h_j H_{j}`, the product formula is constructed by random sampling over the terms
+    of the Hamiltonian. With probability :math:`p_j` we will add to the product the operator
+    :math:`\exp{(\frac{i \lambda H_j}{N})}`, where :math:`\lambda = \sum_{j=1}^{N} |h_j|`.
+    On the other hand, :math:`p_j` is calculated as :math:`p_j = \frac{|h_j|}{\lambda}`.
+    The number of terms to be added to the product will be entered in the class constructor.
+
+    Args:
+        hamiltonian (Union[~.Hamiltonian, ~.Sum]): The Hamiltonian of the system.
+        time (complex): The time for which the system evolves.
+
+    Keyword Args:
+        n (int): The number of terms to be added to the product formula. Default is 1.
```

Comment:
`n` in TrotterProduct is the number of trotter steps. Although it is related, could be confuse have `n` here?  In the paper is called `N`. Also, in the PRD is set default 1 but not sure if that make sense in this context